### PR TITLE
Support surfaces with strides over 2G

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michał Zientkiewicz <mzient@gmail.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes a bug: invalid memory access with large inputs (above 2G elements) (#2588)

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     * Changed strides in `Surface` to 64 bits.
     * Change SampleDesc in warp to use 64-bit strides
 - Affected modules and functionalities:
     * CPU resampling, warp operators
 - Key points relevant for the review:
     * N/A
 - Validation and testing:
     * python test for operator WarpAffine
 - Documentation (including examples):
     * N/A

**JIRA TASK**: DALI-1789

### CODE REVIEW

**Review by jantonguirao — APPROVED** [review]

**Comment by mzient** [comment]
!build

File: dali/test/python/test_operator_warp.py [hunk 1]
```diff
@@ -219,3 +220,45 @@ def test_gpu_vs_cpu():
...
+  out_size = 10
+  channels = 3
+
+  def get_data():
+    out = np.full([in_size, in_size, channels], 42, dtype=np.uint8)
+    for c in range(channels):
+      out[in_size-1, in_size-1, c] = c
+    return [out]
+
+  pipe = Pipeline(1, 3, 0)
```
Comment (JanuszL) [inline dali/test/python/test_operator_warp.py:235]: I would set prefetch_queue_depth to 1 to reduce the number of output buffers.

**Review by JanuszL — APPROVED** [review]

File: dali/test/python/test_operator_warp.py [hunk 2]
```diff
@@ -219,3 +220,45 @@ def test_gpu_vs_cpu():
...
+  input = fn.external_source(source=get_data, device=device)
+
+  rotated = fn.warp_affine(input, matrix=[-1, 0, in_size, 0, -1, in_size],
```
Comment (jantonguirao) [inline dali/test/python/test_operator_warp.py:238]: maybe an np array with an actual matrix? Just a suggestion

File: dali/test/python/test_operator_warp.py [hunk 3]
```diff
@@ -219,3 +220,45 @@ def test_gpu_vs_cpu():
...
+    out, = pipe.run()
+  except RuntimeError as e:
+    if "bad_alloc" in str(e):
+      print("Skipping test due to out-of-memory error:", e)
+      return
+    raise
+  except MemoryError as e:
+    print("Skipping test due to out-of-memory error:", e)
+    return
+  if device == "cpu":
```
Comment (jantonguirao) [inline dali/test/python/test_operator_warp.py:254]: you could write this more compact:
```
out = out.as_cpu().at(0) if device == 'gpu' else out.at(0)
```

File: dali/test/python/test_operator_warp.py (hunk 1 above)
Comment (mzient) [reply dali/test/python/test_operator_warp.py]: Done.

File: dali/test/python/test_operator_warp.py [hunk 4]
```diff
@@ -219,3 +220,45 @@ def test_gpu_vs_cpu():
...
+    out = out.at(0)
+  else:
+    out = out.as_cpu().at(0)
+  assert out.shape == (out_size, out_size, channels)
+  for c in range(channels):
+    assert out[0,0,c] == c
+
+def test_extremely_large_data():
+  for device in ["cpu", "gpu"]:  # cpu is too slow and the failure was not detected anyway
```
Comment (jantonguirao) [inline dali/test/python/test_operator_warp.py:263]: mmm, but you are still running cpu

Comment (mzient) [reply dali/test/python/test_operator_warp.py]: Done.

**Pull request merged**